api-common/src/main/java/io/enmasse/api/auth/ApiHeaderConfig.java

```diff
@@ -18,9 +19,9 @@
             Collections.singletonList("X-Remote-Extra-"));
 
     public ApiHeaderConfig(List<String> userHeaders, List<String> groupHeaders, List<String> extraHeadersPrefix) {
-        this.userHeaders = userHeaders;
-        this.groupHeaders = groupHeaders;
-        this.extraHeadersPrefix = extraHeadersPrefix;
+        this.userHeaders = new ArrayList<>(userHeaders);
```

Comment:
I prefer immutable object whenever I can as it makes the world an easier place to reason about, so in addition to the change you made, I would wrap the new ArrayList in a Collections.unmodifiable, that way someone calling the getter can't modify my data.